### PR TITLE
Fix: `arrow-spacing` allow multi-spaces and line-endings (fixes #3079)

### LINKED ISSUE #3079: arrow-spasing rule warns in multiline case
version: `v1.0.0-rc-1`
message: `Unexpected space after =>     arrow-spacing`
In case:
```js
const test = (argument) =>
  ({ argument })
```
and configuration:
```json
"arrow-spacing": [ 2, { "before": true, "after": true }],
```
warning appears.

### PR DESCRIPTION
**Pull request by mysticatea**
- The `no-multi-spaces` rule would handle multi-spaces.
- The `brace-style` rule would handle line endings around the open bracket.

### CODE REVIEW

File: docs/rules/arrow-spacing.md [hunk 1]
```diff
@@ -12,12 +12,12 @@ This rule normalize style of spacing before/after of arrow functions arrow(`=>`)
...
 ## Rule Details
 
-this rules takes one arguments of structure contains `before` and `after` property
+this rule takes one arguments of structure contains `before` and `after` property
 and each property has bool value.
 
 default configuration is `{ "before": true, "after": true }`.
 
-`true` means there should have **one space** and `false` means **no space**.
+`true` means there should have **spaces** and `false` means **no space**.
```
Comment (IanVS) [inline docs/rules/arrow-spacing.md]: I had to read this a few times to understand, perhaps a slight rewording might help:

```
`true` means there should be **one or more spaces** and `false` means **no spaces**.
```

File: docs/rules/arrow-spacing.md [hunk 2]
```diff
@@ -12,12 +12,12 @@ This rule normalize style of spacing before/after of arrow functions arrow(`=>`)
 
 ## Rule Details
 
-this rules takes one arguments of structure contains `before` and `after` property
+this rule takes one arguments of structure contains `before` and `after` property
```
Comment (IanVS) [inline docs/rules/arrow-spacing.md]: The wording here is a little confusing.  Might I suggest something like:

```
This rule takes an object argument with `before` and `after` properties, each with a Boolean value.
```

File: lib/rules/arrow-spacing.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ module.exports = function(context) {
 
     /**
      * Determines whether spaces before after arrow(`=>`) is satisfy rule.
-     * if before/after value is `true`, there should be one space.
+     * if before/after value is `true`, there should be spaces.
```
Comment (IanVS) [inline lib/rules/arrow-spacing.js]: "spaces" is plural, but in this case it should be simply more than zero.  I believe the most accurate is "one or more spaces".

File: lib/rules/arrow-spacing.js [hunk 4]
```diff
@@ -54,30 +54,24 @@ module.exports = function(context) {
         var tokens = getTokens(node);
         var countSpace = countSpaces(tokens);
 
-        if (rule.before === true) {
-            // should one space before arrow
-            if (countSpace.before > 1) {
-                context.report(tokens.before, "Unexpected space before =>");
-            }
+        if (rule.before) {
+            // should spaces space before arrow
```
Comment (IanVS) [inline lib/rules/arrow-spacing.js]: should be space before arrow

File: lib/rules/arrow-spacing.js [hunk 5]
```diff
@@ -54,30 +54,24 @@ module.exports = function(context) {
...
             }
         }
 
-        if (rule.after === true) {
-            // should one space after arrow
-            if (countSpace.after > 1) {
-                context.report(tokens.after, "Unexpected space after =>");
-            }
+        if (rule.after) {
+            // should spaces after arrow
```
Comment (IanVS) [inline lib/rules/arrow-spacing.js]: should be space(s) after arrow

**Comment by IanVS** [comment]
I hate to make grammar nits, so take them or leave them.

**Comment by mysticatea** [comment]
Thank you for correction of English. I feel happy since I'm not used to English :)

**Comment by nzakas** [comment]
@ianvs we are helping @mysticatea with his English, so kind feedback is helpful. :)

**Comment by nzakas** [comment]
Lgtm

**Comment by IanVS** [comment]
His English is already very good!

**Pull request merged**